### PR TITLE
New method: union (111 B)

### PR DESCRIPTION
**Pull request by BabkinAleksandr**
[Ramda union](http://ramdajs.com/docs/#union)

aleksandrbabkin_aleksandrs-macbook-pro____documents_nanoutils

### CODE REVIEW

File: lib/union/union.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	const func = function() { return 'a' }
+	const func2 = function() { return 'a' }
+	const a = ['1', func, 2, 3, false]
+	const b = [1, 2, 3, false, 4, func, '1', func2]
+
+	// 'a' array itself and then only unique values from 'b' array
+	const result = ['1', func, 2, 3, false, 1, 4, func2]
+
+	expect(union(a, b)).toEqual(result)
+	expect(union(a)(b)).toEqual(result)
```
Comment (Beraliv) [inline lib/union/union.test.js]: add `union(b, a)` and `union(b)(a)` as well

**Comment by BabkinAleksandr** [comment]
Added tests and refactored method: in previous version, if we have duplicate values in first array - them would not be filtered, because it was initial value of `reduce` method:

```js
return arr2.reduce(function(acc, item) {
    return acc.indexOf(item) === -1
      ? acc.concat(item)
      : acc
  }, arr1) // here
```

File: lib/union/index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import _curry2 from '../_internal/_curry2'
 
 export default _curry2(function union(arr1, arr2) {
-  return arr2.reduce(function(acc, item) {
+  return arr1.concat(arr2).reduce(function(acc, item) {
     return acc.indexOf(item) === -1
       ? acc.concat(item)
       : acc
-  }, arr1)
+  }, [])
```
Comment (Beraliv) [inline lib/union/index.js]: what's the reason?

Comment (BabkinAleksandr) [reply lib/union/index.js]: https://github.com/Kelin2025/nanoutils/pull/35#issuecomment-367583126

**Pull request merged**